src/index.ts

```diff
@@ -0,0 +1,57 @@
+import { extendEnvironment, task } from "@nomiclabs/buidler/config";
+import { BuidlerPluginError } from "@nomiclabs/buidler/internal/core/errors";
+import { BuidlerRuntimeEnvironment } from "@nomiclabs/buidler/types";
+import { applyExtends, loadConfig } from "solhint/lib/config/config-file";
+import { processPath } from "solhint/lib/index";
+
+function getDefaultConfig() {
+  return {
+    extends: ["solhint:default"]
+  };
+}
+
+function getFormatter(formatterName = "stylish") {
+  try {
+    return require(`eslint/lib/formatters/${formatterName}`);
+  } catch (ex) {
+    throw new BuidlerPluginError(
+      "An error has occurred loading the formatter",
+      ex
+    );
+  }
+}
+
+async function getSolhintConfig() {
+  let solhintConfig;
+  try {
+    solhintConfig = await loadConfig();
+  } catch (err) {
+    solhintConfig = getDefaultConfig();
```

Comment:
I think this can lead to unexpected situations.

Suppose a user creates an invalid config file, like `.solhint.json` with content `asd`. `loadConfig` will probably throw, so this line would be executed, and setting the default config.

The correct behavior should be:
1. If no config file exists, use the default config.
2. If a config file exists:
2.1 If it's valid, use it.
2.2 If it's invalid, print an error.